javaslang/src/main/java/javaslang/collection/TreeSet.java

```diff
@@ -505,9 +505,10 @@
         }
     }
 
+    @SuppressWarnings("unchecked")
     @Override
-    public boolean contains(T element) {
-        return tree.contains(element);
+    public boolean contains(Object element) {
+        return tree.contains((T) element);
```

Comment:
Yes.

I've looked into Java 8's `TreeMap` methods `getEntry(Object key)` and `getEntryUsingComparator(Object key)`. (TreeMap is internally used by TreeSet). There they do also unchecked cases like `(K) key` or `(Comparable<? super K>) key`.

Knowing this, the above class casts in this PR are fine, maybe...